### DISCUSSION TITLE
Managing wine runners from other wine manager apps?

### DISCUSSION THREAD

**Discussion by bluesaka111** [bottlesdevs/Bottles#658] [Ideas]
Is there any plans that allow bottles to auto detect installed runners and created instance from other wine managers and directly alter them (using the same winetricks apps that came from other manager)?

**Comment by mirkobrombin** [comment]
Hi! 

### Short answer
No. Sorry but no. 

### Advanced answer
Our bottles are not common prefixes, they have configurations that only work if they are run through Bottles. There is an importer that the user can use to import prefixes from eg Lutris but these are in most cases broken prefixes, as they are not converted to bottles.

Let's go with an example, imagine creating a prefix in Lutris, then enabling DXVK, VKD3D and other configurations. Next you want to manage this prefix via Bottles, so go into the bottle preferences and expect to see the same configurations you did in Lutris, but no it doesn't happen, this is because Bottles like Lutris, POL and others, use their own configuration file ( bottle.yml in the root of the bottle) and this is generated standard whenever I import a non-Bottles prefix. So if you install DXVK in a prefix, Lutris knows it (I think) because it records it in its configuration file and then next time it will propose to uninstall it because it is already installed.

Can't we read third-party configurations? Of course, but it is a hard job .. we cannot keep up with the updates of the other managers, it requires too much effort which unfortunately we do not have.

A few months ago I was working on this feature in a different way, ie I was checking if DXVK, VKD3D and others are installed with a DLL check. Unfortunately, however, I saw development blocked for several reasons:

- DLLs could be more or less, it depends on the type of installation (for example dxgi is not always present)
- Bottles is constantly updated and I have decided to invest my (little) free time on important features (such as versioning, installers (on huge demand) and the dependency manager.

You also ask me if it is possible to use Winetricks on Bottles (on imported prefixes but in general in our bottles). Unfortunately not, our dependency manager works differently and every movement is recorded to make sure everything is reproducible. This is very important for us and for the future of our installers who will be able to understand immediately if they are compatible with a bottle (by reading the dependency tree) or if it is necessary to create a new one.

Let me know if there are any doubts 😁!
